waiter/src/waiter/auth/authentication.clj

```diff
@@ -201,3 +228,110 @@
               handler' (middleware/wrap-merge handler auth-params-map)]
           (handler' request))
         (handler request)))))
+
+(defn process-auth-expires-at-request
+  "Handler to allow a client to update its knowledge of when a user's cookie-based credentials expire.
+   Returns a json response containing the expires-at key containing the expiration time in UTC epoch seconds.
+   Relies on the metadata in the x-waiter-auth cookie."
+  [password {:keys [headers]}]
+  (let [decoded-auth-cookie (get-and-decode-auth-cookie-value headers password)
+        [auth-principal _ auth-metadata] decoded-auth-cookie
+        {:keys [expires-at]} auth-metadata
+        cookie-valid? (decoded-auth-valid? decoded-auth-cookie)
+        sanitized-expires-at (or (when cookie-valid? expires-at) 0)]
```

Comment:
In what situations is it possible to hit this endpoint without a valid cookie?  Should we avoid emitting anything if the cookie is not valid?

Reply:
If a cookie is expired, we should not emit an expired principal.